vivisect/analysis/generic/symswitchcase.py

```diff
@@ -0,0 +1,1307 @@
+
+        # make pointers/numbers from the ptr/offset locations
+        if baseva is not None:
+            vw.makeNumber(idxloc, itemsize)
+        else:
+            vw.makePointer(idxloc)
+
+    # makeNames (done separately because some indexes may be combined)
+    makeNames(vw, jmpva, cases, baseoff)
+
+    # store some metadata in a VaSet
+    vw.setVaSetRow('SwitchCases', (jmpva, jmpva, count) )
+
+    if baseoff is not None:
+        lower += baseoff
+        upper += baseoff
+    
+    vw.setComment(jmpva, "lower: 0x%x, upper: 0x%x" % (lower, upper))
+
+
+def analyzeFunction(vw, fva, timeout=None):
+    '''
+    Function analysis module.
+    This is inserted right after codeblock analysis
+    '''
+    targetNewFunctions(vw, fva)
+
+    lastdynlen = 0
+    dynbranches = vw.getVaSet('DynamicBranches')
+    done = vw.getMeta('analyzedDynBranches')
+    if done is None:
+        done = []
+
+    if not timeout:
+        timeout = vw.config.viv.analysis.symswitchcase.timeout_secs
+        
+    
+    # because the VaSet is often updated during analysis, we have to check to see if there are new 
+    # dynamic branches to analyze.
+    while lastdynlen != len(dynbranches):
+        lastdynlen = len(dynbranches)
+        for jmpva, (none, oprepr, bflags) in list(dynbranches.items()):
+            if bflags & envi.BR_PROC:   # skip calls
+                continue
+
+            try:
+                funcva = vw.getFunction(jmpva)
+                if funcva != fva:
+                    # jmp_indir is for the entire VivWorkspace.  
+                    # we're just filtering to this function here.
+                    # this should be checked again when codeblocks are allowed to 
+                    #   be part of multiple functions.
+                    continue
+
+                if jmpva in done:
+                    logger.info("...skipping 0x%x - already done", jmpva)
+                    continue
+
+                done.append(jmpva)
+                analyzeJmp(vw, jmpva, timeout)
+            except:
+                logger.info('Exception processing SwitchCase in function 0x%x', fva, exc_info=1)
+
+        dynbranches = vw.getVaSet('DynamicBranches')
+    vw.setMeta('analyzedDynBranches', done)
+    # TODO: we need a better way to store changing lists/dicts, that don't show up in the UI.  VaSet would be great, but ugly
+
+def analyzeJmp(vw, jmpva, timeout=None):
+    if vw.getVaSetRow('SwitchCases', jmpva) is not None:
+        logger.info("...skipping 0x%x - SwitchCases already has it?", jmpva)
+        return
+
+    if vw.getVaSetRow('SwitchCases_TimedOut', jmpva) is not None:
+        logger.warning("Reanalyzing TimedOut SwitchCase: 0x%x (timeout=%r secs)", jmpva, timeout)
+        logger.debug("Removing 0x%x from VaSet SwitchCases_TimedOut", jmpva)
+        vw.delVaSetRow('SwitchCases_TimedOut', jmpva)
+
+    sc = SwitchCase(vw, jmpva, timeout)
+    sc.analyze()
+                
+
+def pathGenConvert(pathgen):
```

Comment:
Who uses this?

Reply:
nobody right now.  i can remove this.  this was back when i thought i might unify the two approaches we take to creating codepaths.  i can see a time in the future we'll probably want to do that, but not in this PR.  it's old enough and big enough.